.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,51 @@
+name: PR Build - Unit Tests
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  unit_test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: gradle/wrapper-validation-action@v1
+      # The first JDK set up gets to be "primary".
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1.4.0
+        with:
+          java-version: 8
+      - name: save JAVA_HOME as JDK8 for later usage
+        run: echo "::set-env name=ORG_GRADLE_PROJECT_jdk8::$JAVA_HOME"
+      # The second JDK is not first in the path but does reset JAVA_HOME
+      - name: Set up JDK 1.7
+        uses: actions/setup-java@v1.4.0
+        with:
+          java-version: 7
+      - name: save JAVA_HOME as JDK7 for later usage
+        run: echo "::set-env name=ORG_GRADLE_PROJECT_jdk7::$JAVA_HOME"
+      # Java is installed; document settings
+      - name: log project settings
+        run: |
+          echo "--- Java versions ---"
+          $ORG_GRADLE_PROJECT_jdk7/bin/java -version
+          $ORG_GRADLE_PROJECT_jdk8/bin/java -version
+      - name: correct JAVA_HOME
+        run: echo "::set-env name=JAVA_HOME::$ORG_GRADLE_PROJECT_jdk8"
```

Comment:
If jdk7 was first setup as "primary" and then jdk8, all subsequent steps would be under jd8.  Then would this step to Correct Java Home still be necessary?

Reply:
Using act, I've locally tested this current flow and also a different one with Jdk7 first, then jdk8.  They both run but I'm getting failed tests in both runs.

Reply:
I quickly tested installing java 7 and then java 8; it seems like java 7 was first in the path when I ran `java`. Reversing the order means that java 8 seemed to appear on the path first, so I went with this order.

Reply:
I'm curious what tests failed. It passed for me locally and it appeared to pass on GH as well.

Reply:
Ok, I think I got it now. I ran it again and printed out the path. Your intention was to have jdk8 first on path.

JAVA_HOME and path to jdk are two separate concerns. The JAVA_HOME is an environment variable other java applications in our tests rely on. 

I'm running it again now, with the jdk8 first on path, and I'm getting this exception during the run:

```
 com.newrelic.agent.datastore.DatastoreMetricsTest > testBuildInstanceIdentifier FAILED
|     java.net.UnknownHostException at DatastoreMetricsTest.java:65
|         Caused by: java.net.UnknownHostException at DatastoreMetricsTest.java:65
```

thrown by:

        `final String hostname = InetAddress.getLocalHost().getHostName();`

I'm looking into it and I'm guessing its my machine and not anything to do with the test.

Reply:
> Ok, I think I got it now. I ran it again and printed out the path. Your intention was to have jdk8 first on path.

Yes, precisely. Thanks!